### DISCUSSION TITLE
Question about encoding text prompt in Stable Diffusion XL

### DISCUSSION THREAD

**Discussion by 00why00** [huggingface/diffusers#7971] [Q&A]
Both stable diffusion 2.1 and XL use the second-last output of the text encoder to compute cross-attention in the unet.
But why in pipeline stable diffusion, it uses the last output of the text encoder and has an additional layer norm operation
https://github.com/huggingface/diffusers/blob/70f8d4b488f03730ae3bc11d4d707bafe153d10d/src/diffusers/pipelines/stable_diffusion/pipeline_stable_diffusion.py#L392-L407
which pipeline stable diffusion xl hasn't
https://github.com/huggingface/diffusers/blob/70f8d4b488f03730ae3bc11d4d707bafe153d10d/src/diffusers/pipelines/stable_diffusion_xl/pipeline_stable_diffusion_xl.py#L403-L407

**Comment by asomoza** [comment]
Hi, what's not clear in the code comment?

This was added with the `clip_skip` implementation and it states that if you don't use it, the prompt embeds passed through the LayerNorm layer, so if you use it, it was required to manually pass the prompt embeds to it.

It seems your question is more related to the difference in architecture of each model and probably for that it will be better if you read the paper of each model and understand them.

cc @sayakpaul if there's something else to add.

**Reply by sayakpaul** [reply]
Nope nothing else to add. You nailed the answer.
